oauth2_http/javatests/com/google/auth/oauth2/OAuth2CredentialsTest.java

```diff
@@ -805,15 +782,21 @@ public void refreshIfExpired_refreshesToken() throws IOException {
     assertEquals(1, transportFactory.transport.buildRequestCount--);
   }
 
-  @Test(expected = IllegalStateException.class)
-  public void refresh_temporaryToken_throws() throws IOException {
-    OAuth2Credentials credentials =
-        OAuth2Credentials.newBuilder().setAccessToken(new AccessToken(ACCESS_TOKEN, null)).build();
-    credentials.refresh();
+  @Test
+  void refresh_temporaryToken_throws() {
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          OAuth2Credentials credentials =
+              OAuth2Credentials.newBuilder()
+                  .setAccessToken(new AccessToken(ACCESS_TOKEN, null))
```

Comment:
This portion should better be taken out of the scope. However, since this doesn't change the semantics of the old code, this is fine.